### PR TITLE
CR Allotment Callable Functions

### PR DESCRIPTION
**Pull request by forsajan**

### CODE REVIEW

**Comment by forsajan** [comment]
[CR Allotment Callable Functions](https://trello.com/c/NX30DlYu/140-cr-allotment-callable-functions)

File: functions/src/modules/CR.functions.ts [hunk 1]
```diff
@@ -1,7 +1,14 @@
 import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
 import * as helpers from '../helpers';
 import GoogleSheets from '../services/GoogleSheets';
 import * as _ from 'lodash';
+import * as moment from 'moment';
+
+export enum ISoundQualityReportSheet {
```
Comment (anantakrishna) [inline functions/src/modules/CR.functions.ts]: Sound Quality Report is not related to the Content Reporting. I'd suggest something like
```suggestion
export enum SheetNames {
```

File: functions/src/modules/CR.functions.ts [hunk 2]
```diff
@@ -63,3 +70,149 @@ export const copySubmissionsToProcessing = functions.pubsub
...
+    .filter(item => !item['Status'] && item['List'])
+    .map(item => item['List'])
+    .filter((value, index, self) => self.indexOf(value) === index);
+});
+
+/**
+ * Gets spare files for specified list and languages
+ */
+export const getSpareFiles = functions.https.onCall(
+  async ({ list, language, languages, count }, context) => {
```
Comment (anantakrishna) [inline functions/src/modules/CR.functions.ts]: Allotment form that is going to call this function will be sending only `languages`, no need to maintain backward-compatibility with the old form that was sending only one `language`.

File: functions/src/modules/CR.functions.ts [hunk 3]
```diff
@@ -63,3 +70,149 @@ export const copySubmissionsToProcessing = functions.pubsub
...
+    const fileNameColumn = await sheet.getColumn('File Name');
+    const emailColumn = await sheet.getColumn('Email');
+
+    /// Update files in the Allotments sheet, in parallel
+    await Promise.all(
+      files.map(async file => {
+        const index = fileNameColumn.indexOf(file.filename);
+        if (index < 0) {
+          console.warn(
+            `File ${file.filename} is not found in the SQR allotments.`
```
Comment (anantakrishna) [inline functions/src/modules/CR.functions.ts]: SQR is not relevant here.

**Pull request merged**